src/utils/cookies.ts

```diff
@@ -0,0 +1,61 @@
+import { IS_DOCUMENT } from '@/constants/global-constants';
+import { IncomingMessage } from 'http';
+
+type CookieOptions = {
+  path?: string;
+  secure?: boolean;
+  'max-age'?: number;
+  samesite?: 'strict' | 'lax' | 'none';
+};
+
+export function getCookie(
+  name: string,
+  req?: IncomingMessage,
+): string | undefined {
+  if (IS_DOCUMENT) {
```

Comment:
얼리 리턴으로 뎁스를 하나 줄일 수 있을 것 같아요~

```suggestion
  if (!IS_DOCUMENT) return;
```

Reply:
엇
```
  if (IS_DOCUMENT) {
    if (!req?.headers?.cookie) {
      return undefined;
    }
    const matches = req.headers.cookie.match(
      new RegExp(
        `(?:^|; )${name.replace(/([.$*|{}()[\]\\/+^])/g, '\\$1')}=([^;]*)`,
      ),
    );
    return matches ? decodeURIComponent(matches[1]) : undefined;
  }
  ```
  이 부분의 첫번째 if문 말씀하시는게 맞을까요?? 중첩된 조건문인데 한 번만 확인 부탁드려용!

Reply:
```if (!req?.headers?.cookie) return undefined;```
혹시 두 번째 줄을 이렇게!?

Reply:
`IS_DOCUMENT`가 false면 함수에서 할 일이 더 없을 줄 알았는데 뒤에 있었네요!
이 경우에는 아래처럼 중복을 피할 수 있을 것 같습니다.

```ts
export function getCookie(
  name: string,
  req?: IncomingMessage,
): string | undefined {
  const cookiePattern = new RegExp(
    `(?:^|; )${name.replace(/([.$*|{}()[\]\\/+^])/g, '\\$1')}=([^;]*)`,
  );

  if (req?.headers?.cookie) {
    const matches = req.headers.cookie.match(cookiePattern);
    return matches ? decodeURIComponent(matches[1]) : undefined;
  }

  if (IS_DOCUMENT) {
    const matches = document.cookie.match(cookiePattern);
    return matches ? decodeURIComponent(matches[1]) : undefined;
  }

  return undefined;
}
```

혹시 모르니 적용하신다면 이전 실행이랑 변화가 없는지 확인 한 번 해주세요!